dataland-community-manager/src/main/kotlin/org/dataland/datalandcommunitymanager/services/CommunityManagerListener.kt

```diff
@@ -114,13 +117,82 @@ class DataRequestUploadListener(
+        val dataId = privateDataUploadMessage.dataId
         if (dataId.isEmpty()) {
             throw MessageQueueRejectException("Provided data ID is empty")
         }
         MessageQueueUtils.rejectMessageOnException {
-            dataRequestAlterationManager.patchRequestStatusFromOpenOrNonSourceableToAnsweredByDataId(dataId, correlationId = id)
+            dataRequestUpdateManager.processUserRequests(
+                dataId,
+                correlationId = id,
+            )
+        }
+    }
+
+    /**
+     * Listens for message that specifies a dataset as non-sourceable
+     * and patches all requests corresponding to this dataset to the request status non-sourceable.
+     * @param payload the message describing the result of the data non-sourceable event
+     * @param type the type of the message
+     * @param correlationId the correlation id of the message
+     */
+    @RabbitListener(
+        bindings = [
+            QueueBinding(
+                value =
+                    Queue(
+                        "community-manager.queue.nonSourceableData",
+                        arguments = [
+                            Argument(name = "x-dead-letter-exchange", value = ExchangeName.DEAD_LETTER),
+                            Argument(name = "x-dead-letter-routing-key", value = "deadLetterKey"),
+                            Argument(name = "defaultRequeueRejected", value = "false"),
+                        ],
+                    ),
+                exchange = Exchange(ExchangeName.BACKEND_DATA_NONSOURCEABLE, declare = "false"),
+                key = [RoutingKeyNames.DATA_NONSOURCEABLE],
+            ),
+        ],
+    )
+    fun processDataReportedNonSourceableMessage(
+        @Payload payload: String,
+        @Header(MessageHeaderKey.TYPE) type: String,
+        @Header(MessageHeaderKey.CORRELATION_ID) correlationId: String,
+    ) {
+        MessageQueueUtils.validateMessageType(type, MessageType.DATA_NONSOURCEABLE)
+        val nonSourceableMessage = MessageQueueUtils.readMessagePayload<NonSourceableMessage>(payload, objectMapper)
+
+        if (nonSourceableMessage.receivedDataIsIncomplete()) {
+            throw MessageQueueRejectException("Received data is incomplete")
+        }
+
+        if (!nonSourceableMessage.datasetWasSetToNonSourceable()) {
+            throw MessageQueueRejectException("Received event did not set a dataset to status non-sourceable")
+        }
+
+        logger.info(
+            "Received data-non-sourceable-message for data type: ${nonSourceableMessage.dataType}, " +
+                "company ID: ${nonSourceableMessage.companyId} and reporting period: ${nonSourceableMessage.reportingPeriod}. " +
+                "Correlation ID: $correlationId",
+        )
+
+        val dataTypeDecoded = DataTypeEnum.decode(nonSourceableMessage.dataType)
+
+        if (dataTypeDecoded == null) {
+            throw MessageQueueRejectException("Data type could not be decoded.")
+        }
```

Comment:
Zusammenfassen mit Elvis-Operator

```suggestion
        val dataTypeDecoded = DataTypeEnum.decode(nonSourceableMessage.dataType) ?: throw MessageQueueRejectException("Data type could not be decoded.")
```

Reply:
Done.